### ISSUE TITLE
Python functions with and without type hints displayed inconsistently.

### ISSUE THREAD

**Issue by MarkShackman** [doxygen/doxygen#9846]
Seen in doxygen versions 1.8.18 and 1.9.6, with an unchanged Doxyfile.

Python functions with a type hint are displayed inconsistently to those without a type hint. Specifically, where there is no type hint, 'def' is displayed, but 'def' is not a valid return type. For example:

```
## @file

def docstringWithTypeHintNone() -> None:
 """! comment 1
 """
 pass

def docstringWithTypeHintBool() -> bool:
 """! comment 2
 """
 pass

def docstringWithoutTypeHint():
 """! comment 3
 """
 pass

def docstringWithUnspecifiedTypeHintButShowingWhatTheOutputShouldBe() -> :
 """! comment 4
 """
 pass
```
The first two functions display with the type hint as the function return type. The third example displays with 'def' as the apparent return type. The last example (is invalid Python but)  shows how a function without a type hint should be displayed.

**Comment by albert-github** [comment]
I can reproduce it with a default doxygen settings file but in future please:
- please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

The result up to version 1.8.14 was like:
![image](https://user-images.githubusercontent.com/5223533/218074617-7baaeada-d304-4387-a581-df48e7ee999c.png)

and in the detailed section:

![image](https://user-images.githubusercontent.com/5223533/218075698-56fd8f1e-8571-4127-8c93-74b369b1417f.png)

since 1.8.15 we see:
![image](https://user-images.githubusercontent.com/5223533/218074778-85de5c6d-352e-4794-a70a-bd822f1e3168.png)

and in the detailed section:

![image](https://user-images.githubusercontent.com/5223533/218075810-e34e7cff-89b5-4598-b514-b41df739fc8f.png)

So till 1.8.14 all python functions were shown as functions, in your words, returning "def" which actually means "we don't know what type will be returned.

What would your suggestion be in case there is no type hint given? Just leave it blank?

**Comment by MarkShackman** [comment]
Yes.   Our engineers recommended that, as shown in last example I gave (the one with the long name & comment 4), the return field should be blank.

Thanks.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9848

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
